contracts/DKVDaggerHashimoto.sol

```diff
@@ -195,15 +201,18 @@ contract DecentralizedKVDaggerHashimoto is DecentralizedKV {
 
     /*
      * Run a modified hashimoto hash.
-     * This is eaiser running in evm
+     * This is eaiser running in evm, and it is designed for smaller size
      */
     function _hashimotoKeccak256(
         uint256 startShardId,
         uint256 shardLenBits,
         bytes32 hash0,
         bytes[] memory maskedData
     ) internal view returns (bytes32) {
-        require(maskedData.length == randomChecks, "incorrect PoRA");
+        /* Assumption check */
+        require(maskedData.length == randomChecks, "data vs checks: length mismatch");
+        require(maskedData[0].length <= chunkSize, "too large data uploaded");
```

Comment:
maskedData length should equal to chunkSize so that the masked data has full entropy.